### ISSUE TITLE
Un-report react-test-renderer's findByType & findAllByType queries as async.

### ISSUE THREAD

**Issue by Mrblackey** [testing-library/eslint-plugin-testing-library#671]
### What rule do you want to change?

await-async-query

### Does this change cause the rule to produce more or fewer warnings?

Fewer warnings

### How will the change be implemented?

Check that given node does not end with 'ByType'.

### Example code

	const elements = root.findAllByType('div')

### How does the current rule affect the code?

	promise returned from `findAllByType` query must be handled

### How will the new rule affect the code?

This rule will not be applied if the query ends with 'ByType'.

### Do you want to submit a pull request to change the rule?

Yes

**Comment by Belco90** [comment]
Hi @Mrblackey! This false positive is related to the Aggressive Reporting feature. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

**Comment by Mrblackey** [comment]
Hello.
This option requires us to maintain the custom queries in two places instead of one, or ditch Aggresive Reporting altogether, which I prefer not to.
How about adding a new shared setting: `testing-library/ignore-custom-queries: ["ByType"]`?

**Comment by Belco90** [comment]
> This option requires us to maintain the custom queries in two places instead of one

How so?

> or ditch Aggresive Reporting altogether

That's not necessary, you have granular control over modules and queries that would be reported without opting-out the Aggressive Reporting.

I don't have full context, but if you have a custom query, it should follow the same pattern as the built-in ones. That means that `findBy*` queries are async, and must be reported as such.

**Comment by Mrblackey** [comment]
Assuming we have "findByIcon" query, we add
```
"settings": {
	"testing-library/custom-queries": [
		"findByIcon"
	]
}
```
which is reported when not awaited. This works, as now react-test-renderer's `findByType` is not being reported, while our custom query is. What if we are adding a new custom query tomorrow? Someone has to remember to add it also to the settings section, in order to opt-in this new query to the rule. Perhaps the other way around makes more sense? Maybe assume that each matching query is indeed subjecy to this rule, unless specifically specified otherwise?

**Comment by Belco90** [comment]
Oh, wait. I didn't realize that `react-test-renderer` has its own `findByType` and `findByProps`. That definitely shouldn't be reported by any rule in this plugin (i.e. if the "query" is coming from `react-test-renderer` then should be ignored).

I think that bit is clear. What I don't get is do you actually want to stop reporting a custom `findBy*` query, or the `findByType` from `react-test-renderer`?

> Perhaps the other way around makes more sense? Maybe assume that each matching query is indeed subject to this rule, unless specifically specified otherwise?

That's definitely debatable. It could be interesting to be able to disable custom queries rather than indicating which ones to report (making sure both options can't be enabled at the same time). Happy to receive a PR for that.

**Comment by Mrblackey** [comment]
> I think that bit is clear. What I don't get is do you actually want to stop reporting a custom `findBy*` query, or the `findByType` from `react-test-renderer`?
Only the react-test-renderer ones.

> That's definitely debatable. It could be interesting to be able to disable custom queries with a new option rather than indicating which ones to report (making sure both options can't be enabled at the same time). Happy to receive a PR for that.
So you're basically saying these are two separate things, right? Reports of react-test-renderer should be avoided even without this PR, right?

**Comment by Belco90** [comment]
> So you're basically saying these are two separate things, right? Reports of react-test-renderer should be avoided even without this PR, right?

Yes. I've created #673 to stop reporting utils from `react-test-renderer`.

**Comment by Belco90** [comment]
Taking care of this one one!

**Comment by Belco90** [comment]
Closing in favor of #673
